dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs

```diff
@@ -53,28 +55,29 @@ internal static void ValidFunctionName([NotNull] string? functionName)
 
     internal static void ValidFunctionParamName([NotNull] string? functionParamName)
     {
-        NotEmpty(functionParamName, "The function parameter name cannot be empty");
+        NotNullOrWhiteSpace(functionParamName);
         if (!s_asciiLettersDigitsUnderscoresRegex.IsMatch(functionParamName))
         {
             ThrowInvalidName("function parameter name", functionParamName);
         }
     }
 
-    internal static void StartsWith(string text, string prefix, string message)
+    internal static void StartsWith(string text, string prefix, string message, [CallerArgumentExpression(nameof(text))] string? textParamName = null)
     {
-        NotEmpty(text);
-        NotNull(prefix);
+        Debug.Assert(prefix is not null);
+
+        NotNullOrWhiteSpace(text, textParamName);
         if (!text.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
         {
-            ThrowValidationException(ValidationException.ErrorCodes.MissingPrefix, message);
+            throw new ArgumentException(textParamName, message);
         }
     }
 
     internal static void DirectoryExists(string path)
     {
         if (!Directory.Exists(path))
         {
-            ThrowValidationException(ValidationException.ErrorCodes.DirectoryNotFound, $"Directory not found: {path}");
+            throw new DirectoryNotFoundException($"Directory '{path}' could not be found.");
```

Comment:
nit: I think we could just pass "path" to DirectoryNotFoundException and do the format in the exception class

Reply:
DirectoryNotFoundException is a built-in .NET exception.  Its ctor just takes a message; it doesn't have one that takes a path and wraps that in a message.

Reply:
ah, gotcha!